### PR TITLE
Run SDK End-End Tests

### PR DESCRIPTION
**Pull request by tkatzir**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
related: https://jira-hq.paloaltonetworks.local/browse/CIAC-870

## Description
A few sentences describing the overall goals of the pull request's commits.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

### CODE REVIEW

File: Tests/integration_tests/test_e2e_demisto_sdk_flow_foo.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import random
+
+from e2e_tests_utils import e2e_tests_utils
+
+
+def test_e2e_demisto_sdk_flow_foo(tmpdir, insecure: bool = False):
+    # Importing TestSuite classes from Demisto-SDK
+    e2e_tests_utils.cli(f'mkdir {tmpdir}/git')
```
Comment (dorschw) [inline Tests/integration_tests/test_e2e_demisto_sdk_flow_foo.py:8]: doesn't `tmpdir` create the dir for us? do we really need to `mkdir`?

**Review by dorschw — APPROVED** [review]
nice

File: Tests/integration_tests/test_e2e_demisto_sdk_flow_foo.py (hunk 1 above)
Comment (tkatzir) [reply Tests/integration_tests/test_e2e_demisto_sdk_flow_foo.py:8]: I use `tmpdir` for several things:

- The sdk repo clone destination
- The pack I created for this test

Comment (tkatzir) [reply Tests/integration_tests/test_e2e_demisto_sdk_flow_foo.py:8]: I use tmpdir for several things:

- The sdk repo clone destination
- The pack I created for this test

File: scripts_python_2.txt [hunk 2]
```diff
@@ -0,0 +1,148 @@
+Packs/AbuseDB/Scripts/AbuseIPDBPopulateIndicators/AbuseIPDBPopulateIndicators.yml
```
Comment (yaakovi) [inline scripts_python_2.txt:1]: remove

File: Tests/integration_tests/e2e_tests_utils.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                                               )
+
+
+class e2e_tests_utils:
+
+    @staticmethod
+    def git_clone_demisto_sdk(destination_folder: str, sdk_git_branch: str = 'master'):
+        '''Clone demisto-sdk from GitHub and add it to sys.path
+        '''
+        git_clone_command = ('git'
```
Comment (yaakovi) [inline Tests/integration_tests/e2e_tests_utils.py:16]: check if we can use the `git` py-package

File: Tests/integration_tests/e2e_tests_utils.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+                             f' -C {destination_folder}'
+                             ' clone'
+                             f' -b {sdk_git_branch}'
+                             ' --single-branch --depth 1 https://github.com/demisto/demisto-sdk.git'
+                             )
+        print(f'Cloning demisto-sdk with: {git_clone_command}')
+        e2e_tests_utils.cli(git_clone_command)
+        sys.path.insert(1, f'{destination_folder}/demisto-sdk')
+        # type:ignore [return]
```
Comment (yaakovi) [inline Tests/integration_tests/e2e_tests_utils.py:25]: redundant

File: Tests/integration_tests/e2e_tests_utils.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
+
+import subprocess
+import sys
+
+import demisto_client
+from demisto_sdk.commands.common.tools import (get_demisto_version,
+                                               )
+
+
+class e2e_tests_utils:
```
Comment (yaakovi) [inline Tests/integration_tests/e2e_tests_utils.py:10]: no need for a class

Comment (tkatzir) [reply Tests/integration_tests/e2e_tests_utils.py]: [Done](https://github.com/demisto/content/pull/20373/commits/5aa2261ff79dea6b427eb4870b9e9bd6db87cc96).

File: scripts_python_2.txt (hunk 2 above)
Comment (tkatzir) [reply scripts_python_2.txt]: [Done](https://github.com/demisto/content/pull/20373/commits/f9a9e5e01df8f9bd9c574e76851c1d756e820fde).

File: .gitlab/ci/sdk-nightly.yml [hunk 6]
```diff
@@ -155,6 +155,22 @@ demisto-sdk-nightly:run-commands-against-instance:
     - section_end "Destroy instances"
 
 
+demisto-sdk-nightly:run-end-to-end-tests:
+  tags:
+    - gke # Better when not needing docker-in-docker
+  extends:
+    - .sdk-nightly-schedule-rule-always
```
Comment (yaakovi) [inline .gitlab/ci/sdk-nightly.yml:162]: Do we need `.default-job-settings` too?

File: .gitlab/ci/sdk-nightly.yml [hunk 7]
```diff
@@ -155,6 +155,22 @@ demisto-sdk-nightly:run-commands-against-instance:
...
+  variables:
+    INSTANCE_ROLE: "Server Master"
+  needs: ["create-instances"]
```
Comment (yaakovi) [inline .gitlab/ci/sdk-nightly.yml:165]: SDK nightly has its own `create-instances` job.

**Review by yaakovi — APPROVED** [review]

File: .gitlab/ci/sdk-nightly.yml (hunk 6 above)
Comment (tkatzir) [reply .gitlab/ci/sdk-nightly.yml:163]: Same as `demisto-sdk-nightly:run-commands-against-instance`

**Pull request merged**